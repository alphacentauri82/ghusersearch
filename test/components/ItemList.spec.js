import React from 'react';
import renderer from 'react-test-renderer';
import {
  StyleSheetTestUtils,
} from 'aphrodite';

import ItemList from 'components/ItemList';

function FakeComponent({login}) { // eslint-disable-line
  return <p>{login}</p>;
}

describe('Component: ItemList', () => {

  beforeAll(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('should render components in a list', () => {
    const entities = {
      123: {
        login: 'simon',
      },
      456: {
        login: 'john',
      },
    };
    const component = renderer.create(
      <ItemList
        entities={entities}
        ids={[123, 456]}
        isPending={false}
        component={FakeComponent}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should display a message when no items exist', () => {
    const component = renderer.create(
      <ItemList
        entities={{}}
        ids={[]}
        isPending={false}
        component={FakeComponent}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render a Loading component when isPending', () => {
    const component = renderer.create(
      <ItemList
        entities={{}}
        ids={[]}
        isPending={true}
        component={FakeComponent}
      />
    );
    expect(component).toMatchSnapshot();
  });

});
