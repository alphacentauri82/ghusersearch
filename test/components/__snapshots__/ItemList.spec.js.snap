// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ItemList should display a message when no items exist 1`] = `
<p>
  No items to display
</p>
`;

exports[`Component: ItemList should render a Loading component when isPending 1`] = `
<div
  className="u-flex u-flexJustifyCenter u-flexAlignItemsCenter"
>
  <svg />
</div>
`;

exports[`Component: ItemList should render components in a list 1`] = `
<ul>
  <li
    className="ItemList_item_dgdnyv"
  >
    <p>
      simon
    </p>
  </li>
  <li
    className="ItemList_item_dgdnyv"
  >
    <p>
      john
    </p>
  </li>
</ul>
`;
